### PR TITLE
feat(require-returns): add forceReturnsWithAsync

### PR DESCRIPTION
**Pull request by AndrewLeedham**
Async functions implicitly returning `Promise<void>` is now behind the new `forceReturnsWithAsync` (or `forceRequireReturn`) setting.

fixes: #276

### CODE REVIEW

**Comment by AndrewLeedham** [comment]
@brettz9 Thoughts?

**Comment by brettz9** [comment]
It looked good at my bird's eye view. When I may have a chance, I would like to have a closer look at the tests and the tests that were changed. Just those nits for the moment...

**Comment by brettz9** [comment]
I went ahead and fixed the nits and added back the tests that were removed. While some functionality may cover similar ground, I think it is better to be explicit with different tests, with `forceRequireReturn` and `forceReturnsWithAsync` being of course somewhat different. (And in the case of async arrow functions, we do want to cover that is added back along with the few others.)

**Comment by golopot** [comment]
Since `forceReturnsWithAsync` only affects this rule, it should be moved to options?

**Comment by AndrewLeedham** [comment]
> Since `forceReturnsWithAsync` only affects this rule, it should be moved to options?

Makes sense to me. Implemented.

File: test/rules/assertions/requireReturns.js [hunk 1]
```diff
@@ -134,6 +145,33 @@ export default {
...
+      errors: [
+        {
+          line: 2,
+          message: 'Missing JSDoc @returns declaration.'
+        }
+      ],
+      parserOptions: {
+        ecmaVersion: 8
+      },
+      settings: {
```
Comment (golopot) [inline test/rules/assertions/requireReturns.js]: This should be options. (and other places.)

Comment (brettz9) [reply test/rules/assertions/requireReturns.js]: `forceRequireReturn` wasn't a part of this PR, so those are still there are settings, but my recent commit did add one bad `forceRequiresWithAsync` which I've now amended, thanks!

**Pull request merged**